internal/cmp/lb_profiles.go

```diff
@@ -0,0 +1,129 @@
+// (C) Copyright 2022 Hewlett Packard Enterprise Development LP
+
+package cmp
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/HewlettPackard/hpegl-vmaas-cmp-go-sdk/pkg/client"
+	"github.com/HewlettPackard/hpegl-vmaas-cmp-go-sdk/pkg/models"
+	"github.com/HewlettPackard/hpegl-vmaas-terraform-resources/internal/utils"
+	"github.com/tshihad/tftags"
+)
+
+type loadBalancerProfile struct {
+	lbClient *client.LoadBalancerAPIService
+}
+
+func newLoadBalancerProfile(loadBalancerClient *client.LoadBalancerAPIService) *loadBalancerProfile {
+	return &loadBalancerProfile{
+		lbClient: loadBalancerClient,
+	}
+}
+
+func (lb *loadBalancerProfile) Read(ctx context.Context, d *utils.Data, meta interface{}) error {
+	var lbProfileResp models.GetLBSpecificProfilesResp
+	if err := tftags.Get(d, &lbProfileResp); err != nil {
+		return err
+	}
+
+	lbDetails, err := lb.lbClient.GetLoadBalancers(ctx)
+	if err != nil {
+		return err
+	}
+
+	getlbProfileResp, err := lb.lbClient.GetSpecificLBProfile(ctx, lbDetails.GetNetworkLoadBalancerResp[0].ID, lbProfileResp.ID)
+	if err != nil {
+		return err
+	}
+
+	return tftags.Set(d, getlbProfileResp.GetLBSpecificProfilesResp)
+}
+
+func (lb *loadBalancerProfile) Update(ctx context.Context, d *utils.Data, meta interface{}) error {
+	return nil
+}
+
+func (lb *loadBalancerProfile) Create(ctx context.Context, d *utils.Data, meta interface{}) error {
+	setMeta(meta, lb.lbClient.Client)
+	createReq := models.CreateLBProfile{
+		CreateLBProfileReq: models.CreateLBProfileReq{
+			Name:        d.GetString("name"),
+			Description: d.GetString("description"),
+			ServiceType: d.GetString("service_type"),
+			ProfileConfig: models.LBProfile{
+				ProfileType:            d.GetString("profile_type"),
+				RequestHeaderSize:      d.GetInt("request_header_size"),
+				ResponseHeaderSize:     d.GetInt("response_header_size"),
+				ResponseTimeout:        d.GetInt("response_timeout"),
+				HTTPIdleTimeoutName:    d.GetInt("http_idle_timeout"),
+				FastTCPIdleTimeout:     d.GetInt("fast_tcp_idle_timeout"),
+				ConnectionCloseTimeout: d.GetInt("connection_close_timeout"),
+				HaFlowMirroring:        d.GetBool("ha_flow_mirroring"),
+				CookieMode:             d.GetString("cookie_mode"),
+				CookieName:             d.GetString("cookie_name"),
+				CookieType:             d.GetString("cookie_type"),
+				CookieFallback:         d.GetBool("cookie_fallback"),
+				CookieGarbling:         d.GetBool("cookie_garbling"),
+				SSLSuite:               d.GetString("ssl_suite"),
+			},
+		},
+	}
+
+	if err := tftags.Get(d, &createReq.CreateLBProfileReq); err != nil {
+		return err
+	}
+
+	lbDetails, err := lb.lbClient.GetLoadBalancers(ctx)
+	if err != nil {
+		return err
+	}
+
+	createReq.CreateLBProfileReq.ProfileConfig.ProfileType = "application-profile"
```

Comment:
why are we setting this values ? are these default values ? Can we not set in the resource definition ?

Reply:
done